avocado/plugins/diff.py

```diff
@@ -0,0 +1,387 @@
+                    <tr><td class="diff_chg">Changed</td> </tr>
+                    <tr><td class="diff_sub">Deleted</td> </tr>
+                    </table></td>
+                    <td> <table border="" summary="Links">
+                    <tr><th colspan="2"> Links </th> </tr>
+                    <tr><td>(f)irst change</td> </tr>
+                    <tr><td>(n)ext change</td> </tr>
+                    <tr><td>(t)op</td> </tr>
+                    </table></td> </tr>
+                    </table>"""
+
+                job_diff_html = html_diff.make_file(job1_results,
+                                                    job2_results,
+                                                    fromdesc=job1_id,
+                                                    todesc=job2_id)
+                with open(args.html_output, 'w') as f:
+                    f.writelines(job_diff_html)
+
+                sys.stdout.write('\n--\n%s\n' % args.html_output)
+
+            except Exception as e:
+                LOG.error(e.message)
+                sys.exit(exit_codes.AVOCADO_FAIL)
+
+        if getattr(args, 'open_browser', False):
+            setsid = getattr(os, 'setsid', None)
+            if not setsid:
+                setsid = getattr(os, 'setpgrp', None)
+            inout = file(os.devnull, "r+")
+            cmd = ['xdg-open', args.html_output]
+            subprocess.Popen(cmd, close_fds=True, stdin=inout, stdout=inout,
+                             stderr=inout, preexec_fn=setsid)
+
+    def _validate_filters(self, string):
+        diff_filters = string.split(',')
+        valid_filters = ['all',
+                         'cmdline',
+                         '-cmdline',
+                         'time',
+                         '-time',
+                         'variants',
+                         '-variants',
+                         'results',
+                         '-results',
+                         'config',
+                         '-config',
+                         'sysinfo',
+                         '-sysinfo']
+        add_all = False
+        for diff_filter in diff_filters:
+            if diff_filter in valid_filters:
+                if diff_filter.startswith('-') and 'all' not in diff_filters:
+                    add_all = True
+            else:
+                msg = "Invalid option '%s'" % diff_filter
+                raise argparse.ArgumentTypeError(msg)
+
+        if add_all:
+            diff_filters.append('all')
+
+        return diff_filters
+
+    def _get_job_data(self, jobdir):
+        results_json = os.path.join(jobdir, 'results.json')
+        with open(results_json, 'r') as f:
+            data = json.load(f)
+
+        return data
+
+    def _setup_job(self, job_id):
+        if os.path.isdir(job_id):
+            resultsdir = os.path.expanduser(job_id)
+            job_id = os.path.basename(os.path.normpath(resultsdir))[-7:]
```

Comment:
I think it's better to use `os.path.join(resultsdir, id)` as the directory might be renamed (that's how I noticed it does not work)